js/adapt-offlineStorage-scorm.js

```diff
@@ -1,191 +1,226 @@
+      return (this.suspendDataRestored) ? this.scorm.setSuspendData(dataAsString) : false;
+    }
+
+    getCustomStates() {
+      const isSuspendDataStoreEmpty = _.isEmpty(this.suspendDataStore);
+      if (!isSuspendDataStoreEmpty && this.suspendDataRestored) {
+        return _.clone(this.suspendDataStore);
+      }
 
-      var dataAsString = scorm.getSuspendData();
-      if (dataAsString === "" || dataAsString === " " || dataAsString === undefined) return {};
+      const dataAsString = this.scorm.getSuspendData();
+      if (dataAsString === '' || dataAsString === ' ' || dataAsString === undefined) {
+        return {};
+      }
 
-      var dataAsJSON = JSON.parse(dataAsString);
-      if (!isSuspendDataStoreEmpty && !suspendDataRestored) dataAsJSON = _.extend(dataAsJSON, suspendDataStore);
+      let dataAsJSON = JSON.parse(dataAsString);
+      if (!isSuspendDataStoreEmpty && !this.suspendDataRestored) {
+        dataAsJSON = Object.assign(dataAsJSON, this.suspendDataStore);
+      }
       return dataAsJSON;
-    },
+    }
 
-    getCustomState: function(name) {
-      var dataAsJSON = this.getCustomStates();
+    getCustomState(name) {
+      const dataAsJSON = this.getCustomStates();
       return dataAsJSON[name];
-    },
+    }
 
-    useTemporaryStore: function() {
-      var cfg = Adapt.config.get('_spoor');
+    useTemporaryStore() {
+      const cfg = Adapt.config.get('_spoor');
 
-      if (!scorm.lmsConnected || (cfg && cfg._isEnabled === false)) return true;
+      if (!this.scorm.lmsConnected || (cfg && cfg._isEnabled === false)) return true;
       return false;
-    },
+    }
 
+    getLearnerInfo() {
     /**
      * Returns an object with the properties:
      * - id (cmi.core.student_id)
-     * - name (cmi.core.student_name - which is usually in the format "Lastname, Firstname" - but sometimes doesn't have the space after the comma)
+       * - name (cmi.core.student_name - which is usually in the format
+       *   'Lastname, Firstname' - but sometimes doesn't have the space after the comma)
      * - firstname
      * - lastname
      */
-    getLearnerInfo: function() {
-      var name = scorm.getStudentName();
-      var firstname = "", lastname = "";
-      if (name && name !== 'undefined' && name.indexOf(",") > -1) {
-        //last name first, comma separated
-        var nameSplit = name.split(",");
+      let name = this.scorm.getStudentName();
+      let firstname = '';
+      let lastname = '';
+      if (name && name !== 'undefined' && name.indexOf(',') > -1) {
+        // last name first, comma separated
+        const nameSplit = name.split(',');
         lastname = $.trim(nameSplit[0]);
         firstname = $.trim(nameSplit[1]);
-        name = firstname + " " + lastname;
+        name = `${firstname} ${lastname}`;
       } else {
-        console.log("SPOOR: LMS learner_name not in 'lastname, firstname' format");
+        console.log(`SPOOR: LMS learner_name not in 'lastname, firstname' format`);
```

Comment:
rather than logging a message to the console here, I think we should consider splitting on the space instead

reason I say that is having recently re-read the info on this field in the SCORM 2004 docs I saw there's no requirement in SCORM 2004 to return the name in "lastname, firstname" format - so it would be perfectly valid for the LMS to return "firstname lastname". 

not completely convinced that splitting on the space would work for all possible name formats - but it's better than what we have now...

Reply:
done in https://github.com/adaptlearning/adapt-contrib-spoor/pull/197/commits/67c567d4f25ff44ca371467335c9be606ed51e95